core/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java

```diff
@@ -965,7 +965,20 @@ public static int ymdToUnixDate(int year, int month, int day) {
     return julian - EPOCH_JULIAN;
   }
 
+  /** Calculates the Julian Day Number for any date on the Gregorian. The paper of
+   * http://www.cs.utsa.edu/~cs1063/projects/Spring2011/Project1/jdn-explanation.html
+   * provides a brief explanation for this algorithm. */
   public static int ymdToJulian(int year, int month, int day) {
+  	if (month < 1 || month > 12) {
+  		throw new IllegalArgumentException(
+  			"Month of any date on the Gregorian should between 1 to 12, but " + month);
+    }
```

Comment:
We should add some negative tests.

Reply:
negative tests for addMonths? 
See `checkAddMonths(2019, 9, 1, 2016, 8, 1, -37);`

Reply:
You should fix the throws of the test.

Reply:
Seems the throws come from old calcite since https://github.com/apache/calcite/pull/1890 is not merged.